components/purchase_modal/purchase_modal.tsx

```diff
@@ -284,18 +285,54 @@ function DelinquencyCard(props: DelinquencyCardProps) {
-                    <FormattedMessage
-                        defaultMessage={
-                            'Upon reactivation you will be charged the total owed.  Your bill is calculated at the end of the billing cycle based on the number of enabled users.'
-                        }
-                        id={'cloud_delinquency.cc_modal.disclaimer'}
-                    />
-                    <a onClick={seeHowBillingWorks}>
+                    {Boolean(!props.isCloudDelinquencyGreaterThan90Days) && (
+                        <FormattedMessage
+                            defaultMessage={
+                                'Upon reactivation you will be charged the total owed.  Your bill is calculated at the end of the billing cycle based on the number of enabled users. {seeHowBillingWorks}'
+                            }
+                            id={'cloud_delinquency.cc_modal.disclaimer'}
+                            values={{
+                                seeHowBillingWorks: (
+                                    <a onClick={seeHowBillingWorks}>
+                                        <FormattedMessage
+                                            defaultMessage={
+                                                'See how billing works.'
+                                            }
+                                            id={
+                                                'admin.billing.subscription.howItWorks'
+                                            }
+                                        />
+                                    </a>
+                                ),
+                            }}
+                        />
+                    )}
+                    {Boolean(props.isCloudDelinquencyGreaterThan90Days) && (
                         <FormattedMessage
-                            defaultMessage={'See how billing works.'}
-                            id={'admin.billing.subscription.howItWorks'}
+                            defaultMessage={
+                                'Upon reactivation you will be charged the total owed.  You will be further charged {cost} immediately for the cost of a 1 year subscription according to your currently active user count of {users} users. {seeHowBillingWorks}'
+                            }
+                            id={
+                                'cloud_delinquency.cc_modal.disclaimer_with_upgrade_info'
+                            }
+                            values={{
+                                cost: `$${props.cost}`,
+                                users: props.users,
+                                seeHowBillingWorks: (
+                                    <a onClick={seeHowBillingWorks}>
+                                        <FormattedMessage
+                                            defaultMessage={
+                                                'See how billing works.'
```

Comment:
if the long of the strings are not that long, I'd suggest put all in one line
```suggestion
                                                defaultMessage={'See how billing works.'}
```